### PR TITLE
(Most) UI is now greyscale-colored, in preparation of letting you choose your own UI colors

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
This UI changes Midnight, Slimecore, Plasmafire, Clockwork, and Operative to all use the same base icons, recolored. They don't look exactly the same way they do now, but they all look similar enough (except Clockwork, to an extent).

<details>
<summary>UI themes now (bug that hides navigate button)</summary>

dreamseeker_2022-10-22T04-01-45

dreamseeker_2022-10-22T04-01-30

dreamseeker_2022-10-22T04-01-35

dreamseeker_2022-10-22T04-01-51

dreamseeker_2022-10-22T04-03-05
</details>

## Why It's Good For The Game
Significantly lowers the amount of work needed to add a new UI.

Primarily, this is to allow for user colored UIs. I am separate that PR from this one because this should be test merged, and test merging stuff with preferences is somewhat problematic. Plus, the work needed to make sure the preferences UI for it is perfect (with a preview etc) is going to take some time.

Potentially lifts the requirement to replace Operative in #70654.

Daily dose UI as proof of concept:

## Changelog

:cl:
imageadd: Some UI themes have changed slightly, in preparation for being able to customize them.
/:cl:

### CODE REVIEW

File: code/_onclick/hud/hud.dm [hunk 1]
```diff
@@ -375,13 +366,15 @@ GLOBAL_LIST_INIT(available_ui_styles, list(
...
+/datum/hud/proc/update_ui_style(datum/hud_style/hud_style)
+	var/new_ui_style = hud_style.hud_icon()
+
 	// do nothing if overridden by a subtype or already on that style
 	if (initial(ui_style) || ui_style == new_ui_style)
 		return
 
-	for(var/atom/item in static_inventory + toggleable_inventory + hotkeybuttons + infodisplay + screenoverlays + inv_slots)
-		if (item.icon == ui_style)
+	for(var/atom/movable/screen/item in static_inventory + toggleable_inventory + hotkeybuttons + infodisplay + screenoverlays + inv_slots + listed_actions?.actions + palette_actions?.actions)
```
Comment (Mothblocks) [inline code/_onclick/hud/hud.dm:376]: No this cannot be as anything, it fills with nulls sometimes apparently

**Comment by LemonInTheDark** [comment]
only two colors per ui?

**Comment by rellyreri** [comment]
thanks doc

**Review by Imaginos16 — APPROVED** [review]
Looks good!

**Comment by Mothblocks** [comment]
yeah @LemonInTheDark

**Comment by Wallemations** [comment]
Thanks doc

**Comment by LemonInTheDark** [comment]
> yeah @LemonInTheDark

doesn't that lose a lot of detail? if I recall correctly most use at least 4 for detailing, at least I did for the action dropdown stuff

**Comment by Mothblocks** [comment]
not that i could tell, the only one that really did was clockwork

**Comment by LemonInTheDark** [comment]
image
I was gonna pull up an example but the action dropdown doesn't exist so I can't

**Comment by Mothblocks** [comment]
How do you get those buttons, shift drag or something?

**Comment by LemonInTheDark** [comment]
> How do you get those buttons, shift drag or something?

the small squares come from dragging action buttons around.
what's missing is the 64x32 button you click to show/hide the dropdown

**Comment by Mothblocks** [comment]
Broke UI on Manuel TM, investigating

**Comment by Mothblocks** [comment]
It was an issue that used to be especially caught in unit tests but after refactors was no longer tested

**Comment by LemonInTheDark** [comment]
> It was an issue that used to be especially caught in unit tests but after refactors was no longer tested

moth is wrong this would have never been caught, meposters stay winning

**Comment by Mothblocks** [comment]
Will work on this Sunday

**Comment by Mothblocks** [comment]
I'm exhausted and I hope to get back to this someday sorry

**Pull request closed without merging**